### PR TITLE
Assets,locations,attachments can be created multiple times.

### PR DESCRIPTION
**Pull request by eccles**
Problem:
Successive run of samples will create multiple copies of
identical assets,locations, attachments which might exceed any
quotas.

Solution:
Add utility function that only creates an asset if no assets already
exist with the same display_name (making display_name a unique key).
Likewise for locations. Additionally attachments are only uploaded if
a new asset is created.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: archivist_samples/door_entry/run.py [hunk 1]
```diff
@@ -48,324 +47,246 @@
...
+                "wavestone_ext": "managed",
+            },
+        },
+        attachments=[
+            "gdn_side.jpg",
+        ],
     )
 
 
 def create_doors(doors):
```
Comment (henry739) [inline archivist_samples/door_entry/run.py:270]: Having a method called `doors_create` and a method called `create_doors` isn't super clear - perhaps one or both of them could be renamed to illustrate the different jobs they perform?

Comment (eccles) [reply archivist_samples/door_entry/run.py:270]: renaming doors_create to 'doors_creator' would be acceptable?

File: scripts/samples.sh [hunk 2]
```diff
@@ -95,52 +95,18 @@ else
...
-OPEN="${DOOR_ENTRY} --open"
-${OPEN} "Courts of Justice front door,access_card_1"
-${OPEN} "Courts of Justice front door,access_card_3"
-${OPEN} "Courts of Justice front door,access_card_4"
-${OPEN} "Courts of Justice front door,access_card_0"
-${OPEN} "Courts of Justice front door,access_card_2"
-${OPEN} "Bastille front door,access_card_2"
-${OPEN} "City Hall front door,access_card_2"
-${OPEN} "Gare du Nord apartments side door,access_card_2"
+${TEST_NO_DOOR_ENTRY} python3 -m archivist_samples.door_entry ${ARGS} ${NAMESPACE}
```
Comment (henry739) [inline scripts/samples.sh:98]: What's this doing? It looks like it inserts a colon before this sometimes?

File: archivist_samples/door_entry/run.py (hunk 1 above)
Comment (henry739) [reply archivist_samples/door_entry/run.py:270]: Yeah that sounds slightly better - maybe a comment / docstring above the function too?

File: scripts/samples.sh (hunk 2 above)
Comment (eccles) [reply scripts/samples.sh:98]: yes - it is a shorthand way of not executing something in bash

File: archivist_samples/door_entry/run.py (hunk 1 above)
Comment (eccles) [reply archivist_samples/door_entry/run.py:270]: fixed here and else where

**Review by henry739 — APPROVED** [review]

**Pull request merged**